src/core.c

```diff
@@ -802,6 +808,125 @@ static int find_gpu_user_mode_driver(enum EnGpuDrv gpu_driver, char *user_mode_d
 	return err;
 }
 
+#define CLINFO(dev_id, PARAM, prop) \
+	clGetDeviceInfo(dev_id, PARAM, sizeof(prop), &prop, NULL)
+static int get_gpu_comp_unit (struct pci_dev *dev, uint32_t *comp_unit, char *comp_unit_type)
+{
+	uint8_t ret_cl = 0;
+#if HAS_OpenCL
+	uint8_t ret_domain_nv = 0, ret_bus_nv = 0, ret_dev_nv = 0;
+	cl_uint num_pf, num_ocl_dev, ocl_vendor, amd_gfx_major;
+	uint32_t i, j;
+	cl_platform_id  pf_id;
+	cl_device_id    ocl_dev_id;
+	cl_device_topology_amd topo_amd; // for AMD
+	cl_uint ocl_domain_nv, ocl_bus_nv, ocl_dev_nv; // for NVIDIA
+
+	ret_cl = clGetPlatformIDs(0, NULL, &num_pf); // get number of platform
+	if (ret_cl != CL_SUCCESS || !num_pf)
+	{
+		MSG_ERROR(_("This platform does not support OpenCL"), NULL);
+		return ret_cl;
+	}
+	MSG_DEBUG("Number of OpenCL Platform:\t%d", num_pf);
+
+	for (i = 1; i <= num_pf; i++) // find GPU device
+	{
+		ret_cl = clGetPlatformIDs(i, &pf_id, NULL); // get platform id
+		if (ret_cl)
+		{
+			MSG_ERROR(_("failed to call clGetPlatformIDs"), NULL);
+			return ret_cl;
+		}
+
+		ret_cl = clGetDeviceIDs(pf_id, CL_DEVICE_TYPE_GPU, 0, NULL, &num_ocl_dev); // get number of device
+		if (ret_cl)
+		{
+			MSG_ERROR(_("failed to call clGetDeviceIDs"), NULL);
+			return ret_cl;
+		}
+		MSG_DEBUG("Number of OpenCL Device:\t%d", num_ocl_dev);
+
+		for (j = 1; j <= num_ocl_dev; j++)
+		{
+			ret_cl = clGetDeviceIDs(pf_id, CL_DEVICE_TYPE_GPU, i, &ocl_dev_id, NULL); // get deviceid of opencl
+			if (ret_cl)
+			{
+				MSG_ERROR(_("faild to call clGetDeviceIDs"), NULL);
```

Comment:
Sorry, I made a typo...

Reply:
Yeah, I didn't notice. Don't worry, I'm on it.

Reply:
Thank you, @X0rg :pray:

Reply:
Fixed in c3d6f85b378462b043cdbb2d85259789f4196698.